File: PDOO/Practicas/Prac1/Irrgarten (Jv)/src/irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author adam
 */
public class Weapon { 
    private float power;
    private int uses;
    private Dice dado=new Dice();
    public Weapon (float power, int uses){
    this.power=power;
    this.uses=uses;
    }
    public float attack(){
        float ataque=0;
        if(uses>0){
            uses--;
            ataque=power;
        }
        return ataque;
    }
    public String toString(){
        return "W["+power+","+uses+"]";
    }
    public boolean discard(){
        return dado.discardElement(uses);
    }
}

File: PDOO/Practicas/Prac1/irrgarten(rb)/Weapon.rb
# @file Weapon.rb
# @brief Clase que modela un arma del juego.
# @author Alumno
# @date 2025-10-13
# Documentación en formato YARD.

require_relative 'Dice'

module Irrgarten
  # Representa un arma con una cierta potencia y un número limitado de usos.
  class Weapon
    # Construye un arma.
    # @param p [Integer] potencia del arma.
    # @param e [Integer] usos disponibles inicialmente.
    def initialize(p, e)
      @power = p
      @uses = e
    end

    # Realiza un ataque si quedan usos.
    # @return [Integer] la potencia del arma si hay usos; 0 en caso contrario.
    def attack
      salida = 0
      if (@uses > 0)
        salida = @power
        @uses -= 1
      end
      return salida
    end

    # Representación en cadena del arma.
    # @return [String] con el formato "W[power,uses]".
    def to_s
      return "W[" + @power.to_s + "," + @uses.to_s + "]"
    end

    # Decide si el arma debe descartarse en función de los usos restantes.
    # @return [Boolean] true si se descarta, false en caso contrario.
    def discard
      dado = Dice.new
      return dado.discardElement(@uses)
    end
  end
end
